include/Entities/Entity.hpp

```diff
@@ -55,6 +55,12 @@ class Entity {
   // Called when this entity collides with a wall
   virtual void onTouchWall();
 
+  //virtuals overriden in enemy
+  virtual void setStartPos(sf::Vector2f pos);
```

Comment:
I think these should go in an Enemy class that acts as a subclass of Entity and super class to Enemy1/2

Reply:
But they shouldn't go in Entity. If they did we'd have to make every entity implement them

Reply:
The reason I added those virtuals in Entity because Enemy class inherent Entity and Enemy class needs those methods and the compiler complains when I don't add those virtuals in Entity, do you know the other way around?

Reply:
So far I think Character and Enemy will be the subclass of Entity, do you think there will be anything else? Laser is a separate class that has no superclass.

Reply:
any reason to not make Laser a subclass of Entity? It needs a position and needs to be drawable, which our Entity class is designed to do.

The virtuals should not need to go in Entity. If there were issues calling the Enemy functions maybe it was not casted to Enemy correctly?

Reply:
Indeed I wasn't casting them, let me try to do the following at my current branch
* makes Laser a subclass of Entity
* remove Enemy2
* remove enemy virtuals from Entity